### PR TITLE
[SYCLomatic] Simplify segmented sort algorithms

### PR DESCRIPTION
**Pull request by danhoeflinger**
Remove the two pair sorts algorithm options from segmented sort APIs.  This shouldn't hurt much performance wise at runtime, as most cases will not reach this code, but it will significantly reduce the number of kernels required to JIT compile.  

Must be preceded by merging of https://github.com/oneapi-src/SYCLomatic-test/pull/483, to remove the direct testing of this internal API.

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h [hunk 1]
```diff
@@ -2212,17 +2148,11 @@ segmented_sort_keys(
     dpct::internal::segmented_sort_keys_by_parallel_for_of_sorts(
         ::std::forward<Policy>(policy), keys_in, keys_out, n, nsegments,
         begin_offsets, end_offsets, descending, begin_bit, end_bit);
-  } else if (nsegments < 512) // for loop of parallel sorts when we have a small
-                              // number of total sorts to limit total overhead
+  } else
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h:2151]: This line need format

File: clang/runtime/dpct-rt/include/dpl_extras/algorithm.h [hunk 2]
```diff
@@ -2272,19 +2202,12 @@ segmented_sort_pairs(
         ::std::forward<Policy>(policy), keys_in, keys_out, values_in,
         values_out, n, nsegments, begin_offsets, end_offsets, descending,
         begin_bit, end_bit);
-  } else if (nsegments < 512) // for loop of parallel sorts when we have a small
-                              // number of total sorts to limit total overhead
+  } else
```
Comment (yihwang-nv) [inline clang/runtime/dpct-rt/include/dpl_extras/algorithm.h:2205]: Same as above

**Review by yihwang-nv — APPROVED** [review]
LGTM with comments

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**